core/src/test/scala/unit/kafka/log/UnifiedLogTest.scala

```diff
@@ -1630,6 +1630,20 @@ class UnifiedLogTest {
     assertThrows(classOf[OffsetOutOfRangeException], () => LogTestUtils.readLog(log, 1026, 1000))
   }
 
+  @Test
+  def testFlushingEmptyActiveSegments(): Unit = {
```

Comment:
Is it possible to add another test in `LogLoaderTest`, where, we clean shutdown a Log with empty active segment and reload the log again. Then, we should expect that the recovery code path doesn't declare the segment to be corrupted.

Reply:
Done with other comments. Will work on this one next week.

Reply:
Sounds good.